### ISSUE TITLE
Trigger update beats job on some builds

### ISSUE THREAD

**Issue by jalvz** [elastic/beats#13371]
Currently there is an update-beats job in CI that runs on master and ensures that apm-server can update to latest beats version.

Because it only runs on master, it is easy to miss / forget. This is a proposal to run it on PRs that are likely to affect apm-server, that is when changes are detected on:

`devtools/mage*` and `libbeat/scripts/Makefile` 

That should be enough for now. It is important that that stage runs in parallel with others to not impact significantly build times.

Background in https://github.com/elastic/beats/pull/13239#issuecomment-525311140

cc @jsoriano

**Comment by kuisathaverat** [comment]
we are thinking about implementing a scheduled job that checks for changes on the libbeat daily if there are changes this job would create a PR with the libbeat updated.

**Comment by kuisathaverat** [comment]
we can make a job that checks for changes in libbeat daily, this job would create a new PR with the new libbeats updated on the APM Server

**Comment by kuisathaverat** [comment]
This is the job https://beats-ci.elastic.co/job/elastic+apm-server+master+beats-update/

**Comment by kuisathaverat** [comment]
There is a Job for each branch

![Screenshot 2019-11-26 at 12.54.33.png](https://zube.io/files/elastic/1da82e3e56187abc0d89cfc42dc6a959-screenshot-2019-11-26-at-12-54-33.png)

**Comment by kuisathaverat** [comment]
The job downloads the APM Server code and run the script `./script/jenkins/update-beats.sh` so it is easy to add this check to the current pipeline or on a downstream job to do not affect the main pipeline.

**Comment by kuisathaverat** [comment]
@jalvz this PR https://github.com/elastic/apm-server/pull/2988 add the test to the APM Server Pipeline.

@v1v I think we can close https://github.com/elastic/infra/pull/14132

**Comment by kuisathaverat** [comment]
I've made a couple of PRs to create a new pipeline multibranch job that will be launched on every branch/tag/PR that contains changes in some files, these are the regular expressions:

* "^devtools/mage.*",
* "^libbeat/scripts/Makefile"

Also, it can be triggered by a comment  `/run apm-beats-update`

**Comment by kuisathaverat** [comment]
The job is merged and will run on every PR and merge, it would check if have to run the update test or not , it also can be run by putting the comment `/run apm-beats-update`
